File: src/main/java/alfheim/common/core/asm/AlfheimClassVisitor.java
package alfheim.common.core.asm;

import jdk.internal.org.objectweb.asm.ClassVisitor;
import jdk.internal.org.objectweb.asm.MethodVisitor;
import jdk.internal.org.objectweb.asm.Opcodes;

public class AlfheimClassVisitor extends ClassVisitor {

	public AlfheimClassVisitor(ClassVisitor cv) {
		super(Opcodes.ASM5, cv);
	}
	
	@Override
	public MethodVisitor visitMethod(int access, String name, String desc, String signature, String[] exceptions) {
		if (name.equals("addBindInfo")) return new AlfheimMethodVisitor(super.visitMethod(access, name, desc, signature, exceptions));
		return super.visitMethod(access, name, desc, signature, exceptions);
	}
}

File: src/main/java/alfheim/common/core/asm/AlfheimClassVisitor.kt
package alfheim.common.core.asm

import jdk.internal.org.objectweb.asm.ClassVisitor;
import jdk.internal.org.objectweb.asm.MethodVisitor;
import jdk.internal.org.objectweb.asm.Opcodes;

class AlfheimClassVisitor(val cv: ClassVisitor): ClassVisitor(Opcodes.ASM5, cv) {
    
    override fun visitMethod(access: Int, name: String, desc: String, signature: String, exceptions: Array<String>) : MethodVisitor {
        if (name.equals("addBindInfo")) return AlfheimMethodVisitor(super.visitMethod(access, name, desc, signature, exceptions));
        return super.visitMethod(access, name, desc, signature, exceptions);
    }
}